### PR TITLE
Terminology updates pt1

### PR DESCRIPTION
**Pull request by joshkauffman**
As per @garious instructions, will be submitting many chunked PRs

### CODE REVIEW

File: docs/src/terminology.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ The following terms are used throughout the documentation.
 
 A persistent file addressed by [public key](terminology.md#public-key) and with [lamports](terminology.md#lamport) tracking its lifetime.
 
+## active set
+
+The list of validators that can be considered for the next epoch’s leader schedule
```
Comment (garious) [inline docs/src/terminology.md:13]: can be considered?  also, link to `[validators](terminology.md#validator)`, and end with a period

File: docs/src/terminology.md [hunk 2]
```diff
@@ -16,6 +20,10 @@ A front-end application that interacts with a Solana cluster.
 
 The result of interpreting all programs on the ledger at a given [tick height](terminology.md#tick-height). It includes at least the set of all [accounts](terminology.md#account) holding nonzero [native tokens](terminology.md#native-tokens).
 
+## batches
```
Comment (garious) [inline docs/src/terminology.md:23]: I'm not sure we need this definition. A batch is just a set of something. Looks like you're defining a batch of shreds here, but there's also transaction batches.  Where are you seeing this term exactly?

File: docs/src/terminology.md [hunk 3]
```diff
@@ -16,6 +20,10 @@ A front-end application that interacts with a Solana cluster.
...
+
+To speed up the transmission of information throughout nodes, the information is cut into batches. This allows for smaller segments to be passed around, and then shared amongst peers. Once a node has received all batches, it can reconstruct the original set.
```
Comment (garious) [inline docs/src/terminology.md:25]: This document just provides definitions. Context like "To speed up..." should be in some doc that explains the validator's implementation.

File: docs/src/terminology.md [hunk 4]
```diff
@@ -24,6 +32,10 @@ A contiguous set of [entries](terminology.md#entry) on the ledger covered by a [
 
 A preimage resistant [hash](terminology.md#hash) of the [ledger](terminology.md#ledger) at a given [block height](terminology.md#block-height). Taken from the last [entry id](terminology.md#entry-id) in the slot
 
+## blockstore
+
+Before a block is considered to have reached finality, validators shall keep all valid forks that are generated through receipt of a shred signed by the expected leader for a given slot. Once a validator’s votes have reached max lockout, they may prune anything beyond the root from their blockstore.
```
Comment (garious) [inline docs/src/terminology.md:37]: The blockstore is a local database of recent blocks.

File: docs/src/terminology.md [hunk 5]
```diff
@@ -36,13 +48,17 @@ The first [validator](terminology.md#validator) to produce a [block](terminology
 
 Smallest encrypted chunk of ledger, an encrypted ledger segment would be made of many CBC blocks. `ledger_segment_size / cbc_block_size` to be exact.
 
+## checkpoint
+
+If there are multiple active forks within the blockstore, each point in which two forks meet would be a checkpoint. When a checkpoint is older than the root, it can be pruned from memory.
```
Comment (garious) [inline docs/src/terminology.md:53]: Where are you seeing this? We shouldn't be using the term "checkpoint". If we do, it should be synonymous with "snapshot".

File: docs/src/terminology.md [hunk 6]
```diff
@@ -36,13 +48,17 @@ The first [validator](terminology.md#validator) to produce a [block](terminology
...
+
 ## client
 
 A [node](terminology.md#node) that utilizes the [cluster](terminology.md#cluster).
 
 ## cluster
 
-A set of [validators](terminology.md#validator) maintaining a single [ledger](terminology.md#ledger).
+A set of [validators](terminology.md#validator) who run a Solana blockchain instance with the same genesis.
```
Comment (garious) [inline docs/src/terminology.md:61]: The original definition looks more precise.

File: docs/src/terminology.md (hunk 1 above)
Comment (joshkauffman) [reply docs/src/terminology.md:13]: Correct me if I'm wrong, but is there anything that ensures that every validator is given a slot for each epoch? If, for example, you enable validating through a Playstation 5 and hundreds of thousands of people enable this, would there not be enough slots? Or if another cluster spun up with very short epics, but had lots of validators, could there not be some validators missing from the leader schedule?

File: docs/src/terminology.md (hunk 2 above)
Comment (joshkauffman) [reply docs/src/terminology.md:23]: Was synthesized from here: https://docs.solana.com/cluster/overview#confirming-transactions

File: docs/src/terminology.md (hunk 3 above)
Comment (joshkauffman) [reply docs/src/terminology.md:25]: Understood. I'm sure not all words will want to be pulled into the Solana Docs. 
I was approaching this as someone new to Solana myself, and wanting to understand how Solana is able to be so performant.

File: docs/src/terminology.md (hunk 4 above)
Comment (joshkauffman) [reply docs/src/terminology.md:37]: I agree your definition is much more concise and clear, however doesn't seem to capture: https://docs.solana.com/cluster/overview#confirming-transactions

File: docs/src/terminology.md (hunk 5 above)
Comment (joshkauffman) [reply docs/src/terminology.md:53]: https://docs.solana.com/cluster/managing-forks

File: docs/src/terminology.md (hunk 1 above)
Comment (garious) [reply docs/src/terminology.md:13]: Thanks for that context. How about "The set of non-delinquent validators; each is eligible for a slot in the next epoch's leader schedule."?  @carllin, is that accurate?

File: docs/src/terminology.md (hunk 2 above)
Comment (garious) [reply docs/src/terminology.md:23]: Ah, how about changing that doc to use the term "shreds", which is the term used in the codebase?

File: docs/src/terminology.md (hunk 4 above)
Comment (garious) [reply docs/src/terminology.md:37]: Blockstore is lower level than the turbine block propagation strategy. When shreds are dropped by turbine and the block cannot be reassembled, the validator switches to its "repair" mechanism to retrieve the missing shreds. It gets those shreds from the blockstore held by other validators.

File: docs/src/terminology.md (hunk 5 above)
Comment (garious) [reply docs/src/terminology.md:53]: Okay, thanks. Yes, that's a little different than a snapshot.  In the codebase, we call it a "frozen bank", but in that high-level document, calling it a checkpoint seems fine.  How about a definition of, "The ledger state at a given slot height."?

File: docs/src/terminology.md (hunk 1 above)
Comment (carllin) [reply docs/src/terminology.md:13]: @garious, they can be delinquent, they just have to be staked :)

Currently leader schedule generation (we think of validators in the leader schedule as the "active set"), in `leader_schedule_utils::leader_schedule` does not take into account voting behavior

Comment (garious) [reply docs/src/terminology.md:13]: Thanks @carllin. So is the active set the staked validators or the subset that's in the leader schedule?

Comment (carllin) [reply docs/src/terminology.md:13]: sorry! I should have been more clear.

Looking at these docs, the definition and references to the "active set" is closely related to voting (i.e. you are staked and voting), which is not actually implemented in the code.

If we want to maintain the current definition, where the active set for epoch `X` is the list of candidates for epoch `X+2`'s  leader schedule, then that would just be the staked validators in epoch `X`, which is not necessarily the subset in leader schedule of the epoch `X` as some validators may have unstaked/joined.

**Comment by garious** [comment]
See https://github.com/solana-labs/solana/issues/11710

**Pull request closed without merging**